### PR TITLE
Made non-upstream patch design order aware

### PR DESCRIPTION
**Pull request by vivekrnv**
#### Why I did it

Currently, non upstream patches are applied only after upstream patches.

Depends on https://github.com/sonic-net/sonic-linux-kernel/pull/313. Can be merged in any order, preferably together

#### What I did it

- Non upstream Patches that reside in the sonic repo will not be saved in a tar file bur rather in a folder pointed out by `EXTERNAL_KERNEL_PATCH_LOC`. This is to make changes to the non upstream patches easily traceable.
- The build variable name is also updated to `INCLUDE_EXTERNAL_PATCHES`
- Files/folders expected under `EXTERNAL_KERNEL_PATCH_LOC` 
```
EXTERNAL_KERNEL_PATCH_LOC/
       ├──── patches/
             ├── 0001-xxxxx.patch
             ├── 0001-yyyyyyyy.patch
             ├── .............
       ├──── series.patch
```
- series.patch should contain a diff that is applied on the sonic-linux-kernel/patch/series file. The diff should include all the non-upstream patches.

#### How to verify it

Build the Kernel and verified if all the patches are applied properly

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

### CODE REVIEW

**Comment by vivekrnv** [comment]
@saiarcot895, @keboliu Please review

File: platform/mellanox/non-upstream-patches/README.md [hunk 1]
```diff
@@ -1,31 +1,29 @@
...
+### Include the non upstream patches while building sonic linux kernel
+
+Set `INCLUDE_EXTERNAL_PATCHES=y` using `SONIC_OVERRIDE_BUILD_VARS` to include these changes before building the kernel.
+- Eg: `NOJESSIE=1 NOSTRETCH=1 NOBUSTER=1 make SONIC_OVERRIDE_BUILD_VARS=' INCLUDE_EXTERNAL_PATCHES=y ' target/debs/bullseye/linux-headers-5.10.0-12-2-common_5.10.103-1_all.deb`
+
+### Downloading the non upstream patch tar while building sonic linux kernel
 
-Set `INCLUDE_EXTERNAL_PATCH_TAR=y` using `SONIC_OVERRIDE_BUILD_VARS` to include these changes before building the kernel.
-- Eg: `NOJESSIE=1 NOSTRETCH=1 NOBUSTER=1 make SONIC_OVERRIDE_BUILD_VARS=' INCLUDE_EXTERNAL_PATCH_TAR=y ' target/debs/bullseye/linux-headers-5.10.0-12-2-common_5.10.103-1_all.deb`
+To download a publicly accessible tar of the non-upstream patches with the directory structure explained above, use the `EXTERNAL_KERNEL_PATCH_TAR` variable.
```
Comment (saiarcot895) [inline platform/mellanox/non-upstream-patches/README.md:29]: I think `EXTERNAL_KERNEL_PATCH_TAR` needs to be replaced with `EXTERNAL_KERNEL_PATCH_URL`?

Comment (vivekrnv) [reply platform/mellanox/non-upstream-patches/README.md:29]: Yes, will update

Comment (vivekrnv) [reply platform/mellanox/non-upstream-patches/README.md]: Updated

**Review by saiarcot895 — APPROVED** [review]

**Comment by vivekrnv** [comment]
@liat-grozovik Can we merge this PR?

**Review by liat-grozovik — APPROVED** [review]

**Comment by StormLiangMS** [comment]
@xumia could you help to review this one?

**Pull request merged**